client/src/components/Schedule/TableView.tsx

```diff
@@ -269,4 +280,41 @@ export function TableView({ data, timeZone, isAdmin, courseId, refreshData, stor
   );
 }
 
+const getCourseEventDataForUpdate = (entity: CourseEvent) => {
+  return {
+    dateTime: entity.dateTime,
+    organizerId: entity.organizerId,
+    place: entity.place,
+    special: entity.special,
+    duration: entity.duration,
+  };
+};
+
+const getCourseTaskDataForUpdate = (entity: CourseEvent) => {
+  if (entity.event.type !== 'deadline') {
+    return {
+      type: entity.event.type,
+      studentStartDate: entity.dateTime,
+      taskOwner: { githubId: entity.organizer.githubId },
+      special: entity.special,
+      duration: entity.duration,
+    };
+  }
+
+  return {
+    studentEndDate: entity.dateTime,
+    taskOwner: { githubId: entity.organizer.githubId },
+    special: entity.special,
+    duration: entity.duration,
+  };
```

Comment:
Can we replace this with something that doesn't duplicate code? For example: 
```
const dataForUpdate = {
  studentEndDate: entity.dateTime,
    taskOwner: { githubId: entity.organizer.githubId },
    special: entity.special,
    duration: entity.duration,
};

if (entity.event.type !== 'deadline') {
  dataForUpdate.type = entity.event.type;
}

return dataForUpdate;
```

Reply:
fix